core/services/assets/BlockAssetManager.php

```diff
@@ -142,14 +155,16 @@ public function addEditorScript($handle, array $dependencies = array())
         if($this->assets->has($handle)){
             return $this->assets->get($handle);
         }
-        return parent::addJavascript(
+        $asset = parent::addJavascript(
             $handle,
             $this->registry->getJsUrl(
                 $this->domain->assetNamespace(),
                 $handle
             ),
             $this->addDefaultBlockScriptDependencies($dependencies)
         );
+        $asset->setHasLocalizedData();
```

Comment:
I assumed you'd just be able to chain this call (like what you did in `CoreAssetManager`) removing the need to set another variable (the methods are fluent right?)?  Also, are you calling the wrong method here?  Shouldn't it be `setRequiresTranslation`?